src/components/RequestGroups/RequestGroupTable.vue

```diff
@@ -229,11 +250,32 @@ export default {
       };
       return defaultQueryParams;
     },
+    updateSemesterOptions: function(data) {
+      for (let semester of data.results) {
+        this.semesterData[semester.id] = semester;
+        this.semesterOptions.push({ value: semester.id, text: semester.id });
+      }
+    },
+    onSemesterChange: function(event) {
```

Comment:
Maybe update the signature to be `onSemesterChange: function(semesterId)`? I say that because line 260 I think would be more clear written `let selectedSemesterData = this.semesterData[semesterId];`